kafka/config_tls.go

```diff
@@ -15,5 +20,9 @@ func (c *TLSConfig) SetDefaults() {
 }
 
 func (c *TLSConfig) Validate() error {
+	if (len(c.CaFilepath) > 0 || len(c.CertFilepath) > 0 || len(c.KeyFilepath) > 0) &&
+		(len(c.Ca) > 0 || len(c.Cert) > 0 || len(c.Key) > 0) {
+		return errors.New("if caFilepath or certFilepath or keyFilepath is specified, ca, cert, key must be empty")
+	}
```

Comment:
```suggestion
	if len(c.CaFilepath) > 0 && len(c.Ca) > 0 {
		return fmt.Errorf("config keys 'caFilepath' and 'ca' are both set. only one can be used at the same time")
	}

	if len(c.CertFilepath) > 0 && len(c.Cert) > 0 {
		return fmt.Errorf("config keys 'certFilepath' and 'cert' are both set. only one can be used at the same time")
	}
	
	if len(c.KeyFilepath) > 0 && len(c.Key) > 0 {
		return fmt.Errorf("config keys 'keyFilepath' and 'key' are both set. only one can be used at the same time")
	}
```

For maximum flexibility (e.g. ca can be configured directly, but cert is specified as a mounted file) we should check each of the three properties separately and provide a more specific error message in this case.